### ISSUE TITLE
API should have an option to wait for a custom action

### ISSUE THREAD

**Issue by sguptatw** [getgauge/taiko#125]
**Proposed behavior**
There should be an API to wait for a custom action

**Current behavior**
Wait for is available only for a given time period. Though wait is implicit in page load and ajax calls this may not be sufficient sometimes

**Steps to replicate**
* Run the script
```
(async () => {
    try {
        await openBrowser({headless:false});
        await goto("thetrainline.com",{timeout:20000});
 
        await write("London",into(inputField(toRightOf("From"))));
        await press("Enter");
        await write("Manchester");
        await press("Enter");
        await click("Tomorrow")
        await click(button({"type":"submit"}));
        await waitFor("2000")
        await click("Continue");
    } catch (e) {
        console.error(e);
    } finally {
        closeBrowser();
    }
 })();
```
> after`await click(button({"type":"submit"}));` there is a page navigation. After the page is loaded, the application selects a default ticket. Then the continue button can be clicked successful. Now the user waits for this manually before selecting the continue button.

However in the script if I do not use waitFor(<time>) then I get an application error that ticket needs to be selected before Continue. If I put a waitFor(<time>) this can lead to flakyness.

An API `waitFor(someCondition)` would be useful here.

**Version**
commit - a5da76d97ef5a1e6ca17534424747b96003c8b13

**Comment by NivedhaSenthil** [comment]
Example:
```
await openBrowser();
await goto("https://codepen.io/saothienhat/pen/mAYpdv");
await click($("#show-modal"));
await waitFor(async () => !(await $("loading-text").exists()));
```

Where we wait for loading to not exists.

**Comment by Apoorva-GA** [comment]
Fix should be available in commit : b66b08fedf93f2ad84c6c271eac8f9be99bba714

**Comment by Debashis9012** [comment]
This issue has been verified and found fixed with commit version ```b66b08f```.

Kindly find the below script with ```waitFor``` custom action
```
(async () => {
    try {
        await openBrowser({headless:false});
        await goto("thetrainline.com",{timeout:20000});

        await write("London",into(inputField(toRightOf("From"))));
        await press("Enter");
        await write("Manchester");
        await press("Enter");
        await click("Tomorrow")
        await click(button({"type":"submit"}));
        //await waitFor("2000")
        waitFor(async () => !(await $("Continue").exists()));
        await click("Continue");
    } catch (e) {
        console.error(e);
    } finally {
        closeBrowser();
    }
 })();
```
